.github/workflows/lint.yml

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+name: Lint all codez
+on:
+  push:
+    tags: 'release-*'
+    branches:
+      - dev
+      - release
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  go:
+    name: Lint go files
+    runs-on: ubuntu-20.04
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.13.6
+      id: go
+
+    - name: install crlfmt
+      run: |
+        # Workaround to download the specific version we use internally
+        cat > go.mod << EOF
+        module vectorized
+        require (
+          github.com/cockroachdb/crlfmt v0.0.0-20200309141329-97b747f03b21
+        )
+        go 1.13
+
+        EOF
+
+        go mod download
+        go install github.com/cockroachdb/crlfmt
+        rm -rf go.mod
+
+    - name: lint go files
+      run: |
+        find . -name *.go -type f | xargs -n1 crlfmt -wrap=80 -w
+        git diff --exit-code
+
+  js:
```

Comment:
It's not a big deal, but I wonder if we should combine all these jobs into a single job so that the PR checks have a single entry for `Check code formatting` instead of one entry for each language type. That list is starting to get pretty large. If we build a docker image as part of the job that contains all these dependencies and cache that docker image then the entire run might be even more snappy.

Reply:
The first reason I went with separate jobs is concurrency, so we can get different and quick feedback on any failures.
Another reason, that I saw while building this action, was that some of those scripts didn't have a builtin diffing/erroring-on-failure mechanism so I had to format files in place and `git diff` to get the needed behaviour.
To do this sequentially in the same environment we would either need to exit on the first erroring file-type or `checkout` the pristine files before each step (which is a bit of a smell).

I don't have any strong opinions to be honest. I think the best way to tackle this in the long run would be a standalone container with all the tools we use, which would fmt everything in its entrypoint. That could be used as is both in the contributors workstation, this action, a  vtools command and builds.